modules/kubectl/init.zsh

```diff
@@ -0,0 +1,32 @@
+#
+# Provides 'kubectl' aliases and utiities.
+#
+# Authors:
+#   Bruno Miguel Custodio <brunomcustodio@gmail.com>
+#
+
+# Return if requirements are not found.
+if (( ! $+commands[kubectl] )); then
+  return 1
+fi
+
+# Enabled completion for 'kubectl'.
+source <(kubectl completion zsh)
```

Comment:
Couple of things:
* We'll need to source this conditionally _iff_ zsh completion for `kubectl` hasn't been loaded already. For example, in homebrew-ed kubernetes-cli, `_kubectl` is provided by default.
* In case, we really have to source this, we'll have to _cache_ this upfront so that we don't perform the costly operation each time zsh is loaded.